promql/engine.go

```diff
@@ -689,19 +687,40 @@ func (ng *Engine) populateSeries(ctx context.Context, querier storage.Querier, s
 				hints.End = hints.End - offsetMilliseconds
 			}
 
-			set, wrn, err = querier.Select(false, hints, n.LabelMatchers...)
-			warnings = append(warnings, wrn...)
-			if err != nil {
-				level.Error(ng.logger).Log("msg", "error selecting series set", "err", err)
-				return err
-			}
-			n.UnexpandedSeriesSet = set
-
+			sp := &storage.SelectParam{SortSeries: false, Hints: hints, Matchers: n.LabelMatchers}
+			selectParams = append(selectParams, sp)
+			selectCalls[sp] = n
 		case *parser.MatrixSelector:
 			evalRange = n.Range
 		}
 		return nil
 	})
+	switch q := querier.(type) {
+	case storage.BatchQuerier:
```

Comment:
You're still special casing here, the engine should always be doing the same thing no matter what storage it is being passed.

Reply:
I see. I'll fix it right away

Reply:
done.

code /promql/
![image](https://user-images.githubusercontent.com/9583245/78676515-1e6fad00-7919-11ea-9800-f6148732e238.png)
code /fanout/
![image](https://user-images.githubusercontent.com/9583245/78676568-31827d00-7919-11ea-90ad-03826cc3ad5b.png)